redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -28,128 +28,284 @@ import app.cash.redwood.lazylayout.widget.RefreshableLazyList
 import app.cash.redwood.ui.Margin
 import app.cash.redwood.widget.MutableListChildren
 import app.cash.redwood.widget.Widget
+import kotlin.math.max
+import kotlinx.cinterop.CValue
 import kotlinx.cinterop.ObjCClass
+import kotlinx.cinterop.readValue
+import kotlinx.cinterop.useContents
+import platform.CoreGraphics.CGRect
+import platform.CoreGraphics.CGRectZero
+import platform.CoreGraphics.CGSize
+import platform.CoreGraphics.CGSizeMake
 import platform.Foundation.NSIndexPath
 import platform.Foundation.classForCoder
+import platform.UIKit.UICollectionView
+import platform.UIKit.UICollectionViewCell
+import platform.UIKit.UICollectionViewDataSourceProtocol
+import platform.UIKit.UICollectionViewDelegateFlowLayoutProtocol
+import platform.UIKit.UICollectionViewFlowLayout
+import platform.UIKit.UICollectionViewLayout
+import platform.UIKit.UICollectionViewScrollDirection
 import platform.UIKit.UIControlEventValueChanged
+import platform.UIKit.UIEdgeInsetsMake
+import platform.UIKit.UILabel
 import platform.UIKit.UIRefreshControl
-import platform.UIKit.UITableView
-import platform.UIKit.UITableViewCell
-import platform.UIKit.UITableViewCellStyle
-import platform.UIKit.UITableViewCellStyle.UITableViewCellStyleDefault
-import platform.UIKit.UITableViewDataSourceProtocol
-import platform.UIKit.UITableViewDelegateProtocol
+import platform.UIKit.UIScrollView
 import platform.UIKit.UIView
+import platform.UIKit.indexPathForRow
 import platform.UIKit.item
 import platform.darwin.NSInteger
 import platform.darwin.NSObject
 
-private const val reuseIdentifier = "cell"
+internal interface Items: Widget.Children<UIView> {
+  var itemsBefore: Int
+  var itemsAfter: Int
+  var isFinishedInitialLoad: Boolean
+  fun itemForGlobalIndex(index: Int): Widget<UIView>
+  fun itemCount(): Int
+}
 
 internal open class UIViewLazyList() : LazyList<UIView> {
 
-  private val itemsList = mutableListOf<Widget<UIView>>()
-
+  // Data
   override val placeholder: Widget.Children<UIView> = MutableListChildren()
 
-  override val items: Widget.Children<UIView> = object : Widget.Children<UIView> {
+  override val items: Items = object : Items {
+
+    override var itemsBefore: Int = 0
+    override var itemsAfter: Int = 0
+    override var isFinishedInitialLoad: Boolean = false
+
+    private val viewPortList = mutableListOf<Widget<UIView>>()
+
+
     override fun insert(index: Int, widget: Widget<UIView>) {
-      itemsList.add(index, widget)
-      tableView.reloadData()
+      println("🟤 insert $index")
+      viewPortList.add(index, widget)
+
+      if (isFinishedInitialLoad) {
+        val indexPath = globalIndexPathForViewportIndex(index)
+       // collectionView.reloadItemsAtIndexPaths(indexPaths = listOf(indexPath))
+      }
+
+      // TODO: We need a signal that the items are done with their initial load..
```

Comment:
Is the idea that the contents of the LazyList will always be paginated with a strict page size?
If so, we could enforce that here--otherwise, it's pretty hard to tell if our "initial load" is done in general.
We could maybe do something along the lines of asking "do the currently visible cells fill the available frame for the view, or is there a gap at the bottom", but that feels like guesswork.